app/Http/Controllers/UserController.php

```diff
@@ -1,10 +1,65 @@
 <?php
 
+declare(strict_types=1);
+
 namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
+use App\Http\Requests\StoreUser;
+use App\Http\Requests\UpdateUser;
+use App\Services\RegistrationService;
+use App\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
 
 class UserController extends Controller
 {
-    //
+    /**
+     * Lists all resources.
+     */
+    public function index(): JsonResponse
+    {
+        $users = User::all();
+
+        return response()->json($users, Response::HTTP_OK);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(StoreUser $request, RegistrationService $service): JsonResponse
+    {
+        $response = $service->registerUser($request->validated());
```

Comment:
AFAIK `$registerUser()` returns `User` instance and `User` is not a response. Variable name is misleading.